### PR TITLE
feat(app): add migration job

### PR DESCRIPTION
**Pull request by StewartJingga**

### CODE REVIEW

**Review by snorlacs — APPROVED** [review]

File: stable/app/values.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@
 replicaCount: 1
 
 image:
-  repository:
+  repository: ~
```
Comment (rohilsurana) [inline stable/app/values.yaml:8]: What is the reason for this change?

File: stable/app/templates/migration-job.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      containers:
+        - name: {{ .Chart.Name }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          command:
+            {{- toYaml .Values.migration.command | nindent 12 }}
+          args:
+            {{- toYaml .Values.migration.args | nindent 12 }}
```
Comment (rohilsurana) [inline stable/app/templates/migration-job.yaml:28]: In the deployment template we are using `entryPoint` instead of `args`.
I think we should refactor all to use same even if it requires a breaking change.

File: stable/app/values.yaml (hunk 1 above)
Comment (StewartJingga) [reply stable/app/values.yaml:8]: just to explicitly show null value. And also if for some reason below lines are accidentally indented, `repository` won't be treated as an object.

File: stable/app/templates/migration-job.yaml (hunk 2 above)
Comment (StewartJingga) [reply stable/app/templates/migration-job.yaml:28]: agreed, should we keep the changes to follow kubernetes fields or back to `entrypoint`? IMHO entrypoint could be misleading as the `args`.

Also it seems like only `guardian` chart is using this base chart.
cc: @rahmatrhd

Comment (rohilsurana) [reply stable/app/templates/migration-job.yaml:28]: Let's go with `args`. Since `guardian` is version locking the dependency, we can make this breaking change without affecting it.
We will also need to make a breaking change on `guardian` chart as well.

File: stable/app/values.yaml (hunk 1 above)
Comment (rohilsurana) [reply stable/app/values.yaml:8]: Got it. Thanks.

**Comment by AkarshSatija** [comment]
Closed #21
